contributing/architecture-eval-lifecycle.md

```diff
@@ -0,0 +1,326 @@
+# Architecture: Evaluations into Allocations
+
+The [Scheduling Concepts][] docs provide an overview of how the scheduling
+process works. This document is intended to go a bit deeper than those docs and
+walk thru the lifecycle of an Evaluation from job registration to running
+Allocations on the clients. The process can be broken into 4 parts:
+
+* [Job Registration](#job-registration)
+* [Scheduling](#scheduling)
+* [Deployment Watcher](#deployment-watcher)
+* [Client Allocs](#client-allocs)
+
+## Job Registration
+
+Creating or updating a Job is a _synchronous_ API operation. By the time the
+response has return to the API consumer, the Job and an Evaluation for that Job
+have been written to the state store of the server nodes.
+
+* Note: parameterized or periodic batch jobs don't create an Evaluation at
+  registration, only once dispatched.
+* Note: The scheduler is very fast! This means that once the CLI gets a
+  response it can immediately start making blocking queries to get information
+  about the next steps, but those are happening _asynchronously_.
```

Comment:
```suggestion
* Note: The scheduler is very fast! This means that once the CLI gets a
  response it can immediately start making queries to get information
  about the next steps.
```

Do we need to know the async details about how that works, here?

Reply:
Yeah good call, that's minutia.